frontend/src/Editor/Components/NumberInput.jsx

```diff
@@ -6,10 +6,13 @@ export const NumberInput = function NumberInput({
   styles,
   setExposedVariable,
   component,
+  darkMode,
   fireEvent,
 }) {
   const { visibility, borderRadius } = styles;
 
+  const textColor = darkMode && styles.textColor === '#000' ? '#fff' : styles.textColor;
```

Comment:
As the default color is **#232e3c**, that should be handled, or else on switching from light to dark the input is not legible
```suggestion
  const textColor = darkMode && ['#232e3c', '#000000ff'].includes(styles.textColor) ? '#fff' : styles.textColor;
```